apps/meteor/ee/server/models/raw/WorkspaceCredentials.ts

```diff
@@ -0,0 +1,60 @@
+import type { RocketChatRecordDeleted, IWorkspaceCredentials } from '@rocket.chat/core-typings';
+import type { IWorkspaceCredentialsModel } from '@rocket.chat/model-typings';
+import type { Collection, Db, Filter, IndexDescription } from 'mongodb';
+
+import { BaseRaw } from '../../../../server/models/raw/BaseRaw';
+
+export class WorkspaceCredentialsRaw extends BaseRaw<IWorkspaceCredentials> implements IWorkspaceCredentialsModel {
+	constructor(db: Db, trash?: Collection<RocketChatRecordDeleted<IWorkspaceCredentials>>) {
+		super(db, 'workspace_credentials', trash);
+	}
+
+	protected modelIndexes(): IndexDescription[] {
+		return [{ key: { _id: 1, scopes: 1, expirationDate: 1, accessToken: 1 }, unique: true }];
+	}
+
+	getCredentialByScope(scope: string): Promise<IWorkspaceCredentials | null> {
+		const query: Filter<IWorkspaceCredentials> = {
+			scopes: {
+				$all: [scope],
+				$size: 1,
+			},
+		};
+
+		return this.findOne(query);
+	}
+
+	async unsetCredentialByScope(scope: string): Promise<void> {
+		const query: Filter<IWorkspaceCredentials> = {
+			scopes: {
+				$all: [scope],
+				$size: 1,
+			},
+		};
+
+		await this.deleteOne(query);
+	}
+
+	async updateCredentialByScope(scope: string, accessToken: string, expirationDate: Date): Promise<void> {
+		const record = {
+			$set: {
+				scopes: [scope],
+				accessToken,
+				expirationDate,
+			},
+		};
+
+		const query: Filter<IWorkspaceCredentials> = {
+			scopes: {
+				$all: [scope],
+				$size: 1,
+			},
+		};
+
+		await this.updateOne(query, record, { upsert: true });
```

Comment:
```suggestion
		return this.updateOne(query, record, { upsert: true });
```
I know we're not using the returned value here, but it looks like a better practice _to me_ to return it so we're ready to use it if we ever to need to validate something. Same for all other methods in this file
I'd like to hear what others think about this, though

Reply:
I agree, I don't think I should remove the return value just because I am not using it now, I'll add it, thanks

Reply:
That's a very good approach @matheusbsilva137 < 3